Promise.fold() (#47)
evaera/roblox-lua-promise: lib/README.md

      returns: Promise<number>
      static: true

    - name: fold
      since: unreleased
      desc: |

>         Folds an array into a promise or a value. The array is traversed sequentially and

Reviewer: Should be "Folds an array of values or promises into a single value", I think!

>         The reducer function can return a promise or directly a value. It always received the last resolved value.

Reviewer: ```suggestion
        The reducer function can return a promise or value directly. Each iteration receives the resolved value from the previous, and the first receives your defined initial value.
```

        The folding will stop at the first rejection encountered.

>         ```

Reviewer: There's no example here, but this looks like the start of a code block. Could you include one?

      params:
        - name: list

>           type: "array<T>"

Reviewer: ```suggestion
          type: "array<T | Promise<T>>"
```

        - name: reducer
          desc: The function to call with the accumulated value, the current element from the array and its index.
          type:
            kind: function
            params: "accumulator: U, value: T, index: number"
            returns: U | Promise<U>
        - name: initialValue
          type: "U"

>       returns: U | Promise<U>

Reviewer: ```suggestion
      returns: Promise<U>
```

This should always return a Promise!
